lib/kitchen/directions/bake_chapter_references/v2.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# The main difference between v1 is that in v2 references are also baked form introduction pages
+module Kitchen::Directions::BakeChapterReferences
+  class V2
+    renderable
+
+    def bake(chapter:, metadata_source:, uuid_prefix: '.', klass: 'references')
+      @metadata = metadata_source.children_to_keep.copy
+      @klass = klass
+      @title = I18n.t(:references)
+      @uuid_prefix = uuid_prefix
+
+      chapter.references.search('h3').trash
+      chapter.non_introduction_pages.each do |page|
+        references = page.references
+        next if references.none?
+
+        references.search('h3').trash
+
+        title = Kitchen::Directions::EocSectionTitleLinkSnippet.v1(page: page)
+        references.each do |reference|
+          reference.prepend(child: title)
+        end
+      end
+
+      chapter.pages(only: :is_introduction?).each do |page|
+        references = page.references
+        next if references.none?
+
+        references.search('h3').trash
+        references.each do |reference|
+          reference.prepend(child:
+            <<~HTML
+              <a href="##{page.title.id}">
+                <h3 data-type="document-title" id="#{page.title.copied_id}">
+                  <span class="os-text" data-type="" itemprop="">#{page.title_text}</span>
+                </h3>
+              </a>
+            HTML
+          )
+        end
+      end
```

Comment:
Instead of two loops, would it work to do something like

```ruby
chapter.pages.each do |page|
  # get the references and trash the h3, etc
  references.each do |reference|
    if page.is_introduction?
       # title a reference from an introduction page
    else
       # title a reference from a non introduction page
    end
  end
end
```

Reply:
@kswanson33 As we discussed on slack the reason for creating two loops instead of one is that when the title is created for pages that are non introduction ones, the numbering of pages also include the intro ones, because they are all counted when they are in one loop:
![image](https://user-images.githubusercontent.com/55203598/121950209-a04b3980-cd59-11eb-91c0-33474fac303c.png)
![image](https://user-images.githubusercontent.com/55203598/121949998-6f6b0480-cd59-11eb-8787-338a6e2af60b.png)
So I'm leaving the two loops here because I couldn't found a solution how to fix this error occurring in one loop iteration.